src/PureCloud.Client/Apis/SocialMediaApi.cs

```diff
@@ -12,80 +11,560 @@ namespace PureCloud.Client.Apis;
+
+        response.EnsureSuccessStatusCode();
+
+        return await response.Content.ReadFromJsonAsync<AsyncQueryStatus>(_options.JsonSerializerOptions, cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public async Task<SocialMediaAsyncDetailQueryResponse> GetAnalyticsMessagesJobResultsAsync(string jobId, string cursor = null, CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(jobId);
+
+        var parameters = new NameValueCollection();
+
+        if (!string.IsNullOrEmpty(cursor))
+        {
+            parameters.Add("cursor", UriHelper.ParameterToString(cursor));
+        }
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var uri = UriHelper.GetUri($"api/v2/socialmedia/analytics/messages/jobs/{Uri.EscapeDataString(jobId)}/results", parameters);
 
-        var response = await _httpClient.GetAsync(uri, cancellationToken);
+        var response = await client.GetAsync(uri, cancellationToken);
 
         response.EnsureSuccessStatusCode();
 
-        return await response.Content.ReadFromJsonAsync<AsyncQueryStatus>(_options, cancellationToken);
+        return await response.Content.ReadFromJsonAsync<SocialMediaAsyncDetailQueryResponse>(_options.JsonSerializerOptions, cancellationToken);
     }
 
     /// <inheritdoc />
     public async Task<AsyncQueryResponse> CreateAnalyticsMessagesJobAsync(SocialMediaAsyncDetailQuery body, CancellationToken cancellationToken = default)
     {
         ArgumentNullException.ThrowIfNull(body);
 
-        var uri = UriHelper.GetUri("/api/v2/socialmedia/analytics/messages/jobs", null);
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
 
-        var response = await _httpClient.PostAsJsonAsync(uri, body, _options, cancellationToken);
+        var uri = UriHelper.GetUri("api/v2/socialmedia/analytics/messages/jobs", null);
+
+        var response = await client.PostAsJsonAsync(uri, body, _options.JsonSerializerOptions, cancellationToken);
 
         response.EnsureSuccessStatusCode();
 
-        return await response.Content.ReadFromJsonAsync<AsyncQueryResponse>(_options, cancellationToken);
+        return await response.Content.ReadFromJsonAsync<AsyncQueryResponse>(_options.JsonSerializerOptions, cancellationToken);
     }
 
     /// <inheritdoc />
-    public async Task DeleteTopicAsync(string topicId, CancellationToken cancellationToken = default)
+    public async Task<bool> DeleteEscalationRuleAsync(string escalationRuleId, CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(escalationRuleId);
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var response = await client.DeleteAsync($"api/v2/socialmedia/escalationrules/{Uri.EscapeDataString(escalationRuleId)}", cancellationToken);
+
+        response.EnsureSuccessStatusCode();
```

Comment:
Remove this line and return response.IsSuccesaStatusCode instead. Do this everywhere we return bool value